### ISSUE TITLE
Pycocotools is killing matplotlib backend

### ISSUE THREAD

**Issue by morawi** [pdollar/coco#433]
Hi, 

I've been having a problem in drawing things via matplotlib.pyplot as its backend is someway/somehow converted into 'agg' (and not TkAgg), which is the one normally used for non-GUI editors. This is a problem that often occurs as shown in lots of stackoverflow, and other Python platforms questions. I've tried many tricks but none of them seem to work. 

Hence, I had to debug all my code to figure out what is the source of the problem as I doubted that either I am messing up the code somewhere and affecting the matplotlib backend, or, there is some package that is affecting it. 

```
from pycocotools.coco import COCO
import matplotlib.pyplot as plt
plt.plot([1,2,4,5])
plt.show()

dump2.py:1: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

```

Now, I am going to restart the `IPython` kernel, and remove  `from pycocotools.coco import COCO`, and all went well.
```

# from pycocotools.coco import COCO
import matplotlib.pyplot as plt

plt.plot([1,2,4,5])
plt.show()

```
![dada](https://user-images.githubusercontent.com/36924479/85230905-fd343e00-b3ea-11ea-9606-9bd70bcc13ad.png)

I am using:
Spyder 4.1.3
IPython 7.13.0
Python 3.7.6
tkinter 8.6
OS: Windows-10 Pro (not sure if the problem exists in Linux and MAC OSs)

I think the solution should be easy and straightforward, even if it is just a Windows-10 related problem.

**Comment by aalzaid** [comment]
Hi morawi,

I have the same problem. Do you have a solution for it.
Thank you

**Comment by morawi** [comment]
@aalzaid 
No, could not solve the problem. I had to save the plot as an image. 

Just replace 

`plt.show()`

with
`plt.savefig(path+'fig_name.jpg')`

Hope this helps

**Comment by bkntr** [comment]
As a bypass, can run:

`from pycocotools.coco import COCO`
`import matplotlib`
`matplotlib.use('TkAgg')`

Note that this needs to be called after importing COCO.

But as a real solution, this line needs to be removed from coco.py:
`import matplotlib; matplotlib.use('Agg')`

**Comment by morawi** [comment]
True, import `matplotlib; matplotlib.use('Agg')` needs to be removed from coco.py 
Funny enough, this is not working at all:

```
import matplotlib
matplotlib.use('Agg')
matplotlib.use('TkAgg')
import matplotlib.pyplot as plt
plt.plot([1,2,4,5])
plt.show()

```
Sadly, what you suggested is not working.

so, maybe it is `matplotlib` to blame.
